python/mach_bootstrap.py

```diff
@@ -132,32 +129,18 @@ def _process_exec(args):
 
 
 def _activate_virtualenv(topdir):
-    virtualenv_path = os.path.join(topdir, "python", "_virtualenv%d.%d" % (sys.version_info[0], sys.version_info[1]))
-    python = sys.executable   # If there was no python, mach wouldn't have run at all!
-    if not python:
-        sys.exit('Failed to find python executable for starting virtualenv.')
-
-    script_dir = _get_virtualenv_script_dir()
-    activate_path = os.path.join(virtualenv_path, script_dir, "activate_this.py")
-    need_pip_upgrade = False
-    if not (os.path.exists(virtualenv_path) and os.path.exists(activate_path)):
-        import imp
-        try:
-            imp.find_module('virtualenv')
-        except ImportError:
-            sys.exit("Python virtualenv is not installed. Please install it prior to running mach.")
-
-        _process_exec([python, "-m", "virtualenv", "-p", python, "--system-site-packages", virtualenv_path])
-
-        # We want to upgrade pip when virtualenv created for the first time
-        need_pip_upgrade = True
-
-    exec(compile(open(activate_path).read(), activate_path, 'exec'), dict(__file__=activate_path))
-
-    python = _get_exec_path(PYTHON_NAMES,
-                            is_valid_path=lambda path: path.startswith(virtualenv_path))
-    if not python:
-        sys.exit("Python executable in virtualenv failed to activate.")
+    virtualenv_path = os.path.join(topdir, "python", "_venv%d.%d" % (sys.version_info[0], sys.version_info[1]))
+    python = sys.executable
+
+    if not in_venv():
+        venv_python_path = os.path.join(
+            virtualenv_path,
+            _get_virtualenv_script_dir(),
+            "python"
+        )
+        if not os.path.exists(virtualenv_path):
+            _process_exec([python, "-m", "venv", "--system-site-packages", virtualenv_path])
+        os.execv(venv_python_path, [venv_python_path] + sys.argv)
```

Comment:
os.execv is known to misbehave on windows (python/cpython#101191, python/cpython#63323). can we use the same approach as activate_this.py, modifying PATH + VIRTUAL_ENV + sys.path + sys.real_prefix + sys.prefix? (see [these](https://stackoverflow.com/a/33637378) [answers](https://stackoverflow.com/a/33637414) for more details)